src/ips/graph_utils.rs

```diff
@@ -123,6 +141,34 @@ pub fn find_lowest_betweenness(nodes_idx: &[usize], state: &IpsState) -> usize {
     lowest_betweenness_idx
 }
 
+/// Create new vector with nodes that have common network type.
+pub fn filter_network(nodes: &[Node], network: NetworkType) -> Vec<Node> {
+    let mut network_nodes = nodes.to_vec();
+
+    // We need to remove nodes that have different network type.
+    // Each iteration we remove one node and then we check if there are any other nodes with
+    // different network type. Then we begin new iteration getting new size of the vector.
+    loop {
+        let mut found = false;
+
+        let len = network_nodes.len();
+        for idx in 0..len {
+            if network_nodes[idx].network_type != network {
+                remove_node(&mut network_nodes, idx);
+                found = true;
+                break;
+            }
+        }
+
+        // If we didn't find any node with different network type we can stop.
+        if !found {
+            break;
+        }
+    }
```

Comment:
maybe an alternative approach (could be a bit faster):

in the first step - with a `for` loop, we find all non-zcash node indices (non-network_type)
```rust
        let mut bastard_indicies = Vec::with_capacity(network_nodes.len());

        // using `rev` so we don't mess up the order of those found indices once we start removing them
        for idx in (0..network_nodes.len()).rev() {          // could be done more idiomatically, probably
            if network_nodes[idx].network_type != network {
                bastard_indicies.push(idx);
            }
        }
```
in the second step - remove all those unwanted indices:
(using `reverse` so we don't mess up the order of those found indices)
```rust
    for undesired_idx in bastard_indicies {
        remove_node(&mut network_nodes, undesired_idx);
    }
```

Reply:
Applied 😃